src/dtx/dtx_resync.c

```diff
@@ -390,3 +390,92 @@ dtx_resync(daos_handle_t po_hdl, uuid_t po_uuid, uuid_t co_uuid, uint32_t ver,
 	ds_cont_child_put(cont);
 	return rc;
 }
+
+struct container_scan_arg {
+	uuid_t	co_uuid;
+	struct dtx_resync_arg *arg;
+};
+
+static int
+container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+		  vos_iter_type_t type, vos_iter_param_t *iter_param,
+		  void *data, unsigned *acts)
+{
+	struct container_scan_arg	*scan_arg = data;
+	struct dtx_resync_arg		*arg = scan_arg->arg;
+	int				rc;
+
+	if (uuid_compare(scan_arg->co_uuid, entry->ie_couuid) == 0) {
```

Comment:
Is it possible?

Reply:
Yes, because it set yield flag in the following, then the iteration will reprobe here, if we do not compare here, the iteration will never end.

Reply:
But the yield will happen after DTX resync has been done against current container, then after the yield, is it possible that the logic will call container_scan_cb() with some former handled container? If yes, how to handle other more earlier containers?

Reply:
Hmm, as I understand, it is impossible for vos_iteration to go backward, unless this is new created container. But in that case, it is ok to process the container twice for DTX resync? 

Anyway this is standard procedure for vos_iterate() callback, see rebuild & aggregation.

Reply:
If that is true, assume the latest handled container is C5 and all its former containers have been handled, then how do you handle if the next vos_iterate() callback run with C4 or C3?